### PR TITLE
ci(gorgone-tests) test zmq standard configuration on debian 11 and alma 8

### PR DESCRIPTION
**Pull request by Evan-Adam**
## Description
Add firsts test on gorgone communication.
for now the test in about zmq in push mode (the historic and default mode).
robot launch 2 gorgone on the same machine and check they communicate by checking the tcp connection is established, and the central see the poller in the internal api response.

**Fixes** # MON-38012

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [X] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>
See github action for tests result, all tests should be green before merging this PR

#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/workflows/gorgone.yml [hunk 1]
```diff
@@ -183,15 +183,29 @@ jobs:
             # this is not the case for debian, and for now I don't know why it was made any different between the 2 Os.
           fi
 
-      - name: Create database
+      - name: Create databases
         run: |
           mysql -h mariadb -u root -ppassword -e "CREATE DATABASE \`centreon\`"
           mysql -h mariadb -u root -ppassword -e "CREATE DATABASE \`centreon-storage\`"
+          mysql -h mariadb -u root -ppassword -e "GRANT ALL PRIVILEGES ON centreon.* TO 'centreon'@'%';"
+          mysql -h mariadb -u root -ppassword -e "GRANT ALL PRIVILEGES ON  \`centreon-storage\`.* TO 'centreon'@'%';"
```
Comment (kduret) [inline .github/workflows/gorgone.yml:191]:
```suggestion
          mysql -h mariadb -u root -ppassword -e "GRANT ALL PRIVILEGES ON centreon.* TO 'centreon'@'%'"
          mysql -h mariadb -u root -ppassword -e "GRANT ALL PRIVILEGES ON  \`centreon-storage\`.* TO 'centreon'@'%'"
```

File: .github/workflows/gorgone.yml [hunk 2]
```diff
@@ -183,15 +183,29 @@ jobs:
...
+
+      - name: Upload gorgone and robot debug artifacts
+        if: failure()
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+        with:
+          name: gorgone-debug-${{ matrix.distrib }}
+          path: |
+            ./log.html
+            /var/log/centreon-gorgone/*/*
+            /etc/centreon-gorgone/*
```
Comment (kduret) [inline .github/workflows/gorgone.yml:207]:
```suggestion
            log.html
            /var/log/centreon-gorgone
            /etc/centreon-gorgone
```

File: centreon-gorgone/packaging/centreon-gorgone.yaml [hunk 3]
```diff
@@ -183,6 +188,7 @@ overrides:
       - perl(DateTime)
       - perl(Try::Tiny)
       - tar
+      - perl-JSON # gorgone_key_thumbprint.pl need the json module, even when json::xs is already installed
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone.yaml:191]: for consistency
```suggestion
      - perl(JSON) # gorgone_key_thumbprint.pl needs the json module, even when json::xs is already installed
```

File: centreon-gorgone/packaging/centreon-gorgone.yaml [hunk 4]
```diff
@@ -213,6 +219,7 @@ overrides:
       - libev-perl
       - libzmq-ffi-perl
       - libclone-choose-perl
+      - libjson-perl # gorgone_key_thumbprint.pl need the json module, even when json::xs is already installed
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone.yaml:222]:
```suggestion
      - libjson-perl # gorgone_key_thumbprint.pl needs the json module, even when json::xs is already installed
```

File: centreon-gorgone/tests/robot/config/includer.yaml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+name: config.yaml
+description: Configuration of centreon-gorgone. Use config.d directory to change configuration
+configuration: !include /etc/centreon-gorgone/@UNIQIDFROMROBOTTESTINCONFIGFILE@/config.d/*.yaml
```
Comment (kduret) [inline centreon-gorgone/tests/robot/config/includer.yaml:3]: some underscores could make the macro more readable :)
ex: `@UNIQ_ID_FROM_ROBOT_TESTING_CONFIG_FILE@`

**Review by kduret — DISMISSED** [review]

**Review by mushroomempires — APPROVED** [review]

**Review by tuntoja — DISMISSED** [review]

**Review by wtermellil — DISMISSED** [review]

**Review by tuntoja — APPROVED** [review]

File: centreon-gorgone/tests/robot/resources/resources.resource [hunk 6]
```diff
@@ -1,27 +1,129 @@
...
+        Copy File    ${file}    /etc/centreon-gorgone/${gorgone_name}/config.d/
+    END
+    ${key_thumbprint}    Run    perl /usr/local/bin/gorgone_key_thumbprint.pl --key-path=/var/lib/centreon-gorgone/.keys/rsakey.priv.pem | cut -d: -f4
+
+    ${result}    Run    sed -i -e 's/@UNIQ_ID_FROM_ROBOT_TESTING_CONFIG_FILE@/${gorgone_name}/g' /etc/centreon-gorgone/${gorgone_name}/includer.yaml
+    
+    ${CMD}    Catenate
+    ...    sed -i -e 's/@KEYTHUMBPRINT@/${key_thumbprint}/g' 
+    ...    -e 's/@DBNAME@/${DBNAME}/g' 
+    ...    -e 's/@DBNAME@/${DBNAME}/g'
```
Comment (omercier) [inline centreon-gorgone/tests/robot/resources/resources.resource:69]:
```suggestion
```

File: centreon-gorgone/tests/robot/resources/resources.resource [hunk 7]
```diff
@@ -1,27 +1,129 @@
...
+        IF    ${response.json()}[data][${poller_id}][ping_failed] > 0 or ${response.json()}[data][${poller_id}][ping_ok] > 0
+            BREAK
+        END
+    END
+    Log To Console    json response : ${response.json()}
+    Should Be True    ${i} < 9    timeout waiting for poller status in gorgone rest api (/api/internal/constatus) : ${response.json()}
+    Should Be True    0 == ${response.json()}[data][${poller_id}][ping_failed]    there was failed ping between the central and the poller ${poller_id}
+    Should Be True    0 < ${response.json()}[data][${poller_id}][ping_ok]    there was no successful ping between the central and the poller ${poller_id}
+
+Setup 2 Gorgone
```
Comment (omercier) [inline centreon-gorgone/tests/robot/resources/resources.resource:110]:
```suggestion
Setup Two Gorgone Instances
```

File: centreon-gorgone/tests/robot/tests/core/push.robot [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+*** Variables ***
+@{process_list}    gorgone_central    gorgone_poller_2
+
+*** Test Cases ***
+test Evan
+    [Teardown]    Stop Gorgone And Remove Gorgone Config    @{process_list}    sql_file=${ROOT_CONFIG}push_db_1_poller_delete.sql
+
+    Log To Console    \nStarting the gorgone setup
+    Setup 2 Gorgone    push_zmq
```
Comment (omercier) [inline centreon-gorgone/tests/robot/tests/core/push.robot:15]:
```suggestion
    Setup Two Gorgone Instances    push_zmq
```

**Review by callapa — APPROVED** [review]

**Review by omercier — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**